src/os/source/vectorsource.js

```diff
@@ -1689,6 +1717,24 @@ os.source.Vector.prototype.getColorModel = function() {
  * @param {os.data.histo.ColorModel} model
  */
 os.source.Vector.prototype.setColorModel = function(model) {
+  let notify = false;
+
+  // Handle when user has manually colored Feature(s) without a ColorModel
+  if (!this.colorModel) {
+    var items = this.getColoredItems();
+    if (items && items.length > 0) {
+      this.setColor(items); // this action clears the values in this.coloredItems_
+      notify = true;
+
+      // WORKAROUND: Notify Map that these features need to be recolored.  This is only called if user
+      // is manually coloring Features without a ColorModel
+      // Events that didn't work: STYLE, FEATURE, HIGHLIGHTED, CHANGED; with and without this.changed()
+      this.setSelectedItems(items); // deselect anything selected, then select (to redraw these)
+      this.setSelectedItems([]); // deselect these
```

Comment:
Try taking a look at what `os.style.notifyStyleChange` does. There is definitely a better way to do this as this will be slow and wipe out a user's selection.